common/djangoapps/entitlements/models.py

```diff
@@ -27,35 +27,59 @@ class CourseEntitlement(TimeStampedModel):
     enrollment_course_run = models.ForeignKey('student.CourseEnrollment', null=True)
     order_number = models.CharField(max_length=128, null=True)
 
+    def to_json(self):
+        run_key = None
+
+        if self.enrollment_course_run is not None:
+            run_key = self.enrollment_course_run.course_id
+
+        return {
+            'username': self.user.username,
+            'uuid': str(self.uuid),
+            'created': str(self.created),
+            'modified': str(self.modified),
+            'course_uuid': str(self.course_uuid),
+            'expired_at': str(self.expired_at),
+            'mode': self.mode,
+            'enrollment_course_run_key': run_key,
+            'order_number': self.order_number,
+        }
+
     @classmethod
-    def entitlements_for_user(cls, user):
+    def get_user_course_entitlement(cls, user, course_uuid, ignore_expired=True):
         """
-        Retrieve all the Entitlements for a User
+        Retrieve The entitlement for the given parent course id if it exists for the User and is active
 
         Arguments:
             user: A Django User object identifying the current user
+            course_uuid(string): The parent course uuid
+            ignore_expired(bool): True if the expired entitlements should be ignored
 
         Returns:
-            All of the Entitlements for the User
+            The single entitlement for the requested parent course id
         """
-        return cls.objects.filter(user=user)
+        if ignore_expired:
+            return cls.objects.filter(user=user, course_uuid=course_uuid, expired_at=None)
+        else:
+            return cls.objects.filter(user=user, course_uuid=course_uuid)
 
     @classmethod
-    def get_user_course_entitlement(cls, user, course_uuid):
+    def get_user_entitlement_enrollment(cls, user, course_uuid):
         """
-        Retrieve The entitlement for the given parent course id if it exists for the User
+        Get the given user enrolled entitlement applied to an enrollment for an active Entitlement
```

Comment:
This wording is pretty dicey. I would change to something more like "Get the given user's active Entitlement that has been applied to an enrollment in the given course"